index.js

```diff
@@ -28,10 +28,31 @@ function apply(options, compiler) {
 
   var runner = runCompilation.bind(this, options);
 
-  compiler.plugin('run', runner);
-  compiler.plugin('watch-run', function onWatchRun(watcher, callback) {
-    runner(watcher.compiler, callback);
-  });
+  if (options.lintDirtyModulesOnly) {
+    var isFirstRun = true;
+    this.startTime = Date.now();
+    this.prevTimestamps = {};
+
+    compiler.plugin('emit', function (compilation, callback) {
+      var dirtyOptions = assign({}, options);
+      var globPatterb = dirtyOptions.files;
+      var changedFiles = getChangedFiles(this, compilation, globPatterb.join('|'));
```

Comment:
Can you pass an object here, naming the parameters? At some point I'll rewrite this plugin in the latest language features and use destructuring. For now it'll make it clearer why these args are being passed in. 

👍🏽 thanks for taking my comments well. Liking the way this is written :)